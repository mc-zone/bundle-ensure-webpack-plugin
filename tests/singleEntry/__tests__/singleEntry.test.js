var path = require("path");
var fs = require("fs");
var vm = require("vm");

describe("singleEntry", () => {
  var bundle = fs.readFileSync(path.resolve(__dirname, "../dist/main.bundle.js"), "utf8");
  var startupScript = fs.readFileSync(path.resolve(__dirname, "../dist/main.startup.js"), "utf8");

  var retryFn = jest.fn();
  var consoleError = jest.fn();
  var ctx;
  beforeEach(() => {
    retryFn.mockClear();
    consoleError.mockClear();
    ctx = { window: { retry: retryFn }, console: { error: consoleError } };
    vm.createContext(ctx);
  });
  test("startup should realize chunk dosen't exist and launch retry", () => {
    vm.runInContext(startupScript, ctx);
    expect(consoleError).toBeCalled();
    expect(retryFn).toBeCalled();
    expect(retryFn.mock.calls[0][0].name).toBe("main");
    expect(retryFn.mock.calls[0][0].filename).toBe("main.bundle.js");
  });

  test("entry shouldn't run immediately but by the startup code", () => {
    expect(() => {
      vm.runInContext(bundle, ctx);
    }).not.toThrow();
    expect(ctx.window.__WP_CHUNKS__).toBeDefined();
    expect(() => {
      vm.runInContext(startupScript, ctx);
    }).toThrow("I am index!");
  });
});
