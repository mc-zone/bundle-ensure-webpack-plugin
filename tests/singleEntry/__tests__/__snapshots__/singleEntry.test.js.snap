// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`singleEntry should trigger retry func again when one reload failed, until out of limit 1`] = `
Object {
  "0": 2,
}
`;
